src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -373,23 +380,25 @@ private void OnVersionMessageReceived(VersionPayload payload)
             SendMessage(Message.Create(MessageCommand.Verack));
         }
 
-        private void RefreshPendingKnownHashes()
+        private void OnTimer()
         {
+            DateTime oneMinuteAgo = TimeProvider.Current.UtcNow.AddMinutes(-1);
             while (pendingKnownHashes.Count > 0)
             {
                 var (_, time) = pendingKnownHashes[0];
-                if (TimeProvider.Current.UtcNow - time <= PendingTimeout)
-                    break;
+                if (oneMinuteAgo <= time) break;
                 pendingKnownHashes.RemoveAt(0);
             }
+            if (oneMinuteAgo > lastSent)
+                EnqueueMessage(Message.Create(MessageCommand.Ping, PingPayload.Create(NativeContract.Ledger.CurrentIndex(Blockchain.Singleton.View))));
```

Comment:
If so, do we still need pong message?   What do you think @roman-khimov  https://github.com/neo-project/neo/issues/1844 , since we use inv-block now.

Reply:
IIUC this particular ping is now sent if there is no message exchange (messages being sent to be precise) happening for a minute with some peer which is useful. Pings in general are to be used for two purposes: detecting dead connections and ensuring we've not missed any new blocks (if we didn't receive any new ones for some time), if that's the way we are after this PR it's perfectly fine.